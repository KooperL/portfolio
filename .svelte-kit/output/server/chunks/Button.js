import {
  c as create_ssr_component,
  h as compute_rest_props,
  g as getContext,
  k as spread,
  l as escape_attribute_value,
  p as escape_object,
} from "./ssr.js"
import { i as is_void } from "./Frame.js"
import { twMerge } from "tailwind-merge"
const Button = create_ssr_component(($$result, $$props, $$bindings, slots) => {
  let $$restProps = compute_rest_props($$props, [
    "pill",
    "outline",
    "size",
    "href",
    "type",
    "color",
    "shadow",
    "tag",
    "checked",
  ])
  const group = getContext("group")
  let { pill = false } = $$props
  let { outline = false } = $$props
  let { size = group ? "sm" : "md" } = $$props
  let { href = void 0 } = $$props
  let { type = "button" } = $$props
  let { color = group ? (outline ? "dark" : "alternative") : "primary" } =
    $$props
  let { shadow = false } = $$props
  let { tag = "button" } = $$props
  let { checked = void 0 } = $$props
  const colorClasses = {
    alternative:
      "text-gray-900 bg-white border border-gray-200 hover:bg-gray-100 dark:bg-gray-800 dark:text-gray-400 hover:text-primary-700 focus-within:text-primary-700 dark:focus-within:text-white dark:hover:text-white",
    blue: "text-white bg-blue-700 hover:bg-blue-800 dark:bg-blue-600 dark:hover:bg-blue-700",
    dark: "text-white bg-gray-800 hover:bg-gray-900 dark:bg-gray-800 dark:hover:bg-gray-700",
    green:
      "text-white bg-green-700 hover:bg-green-800 dark:bg-green-600 dark:hover:bg-green-700",
    light:
      "text-gray-900 bg-white border border-gray-300 hover:bg-gray-100 dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:hover:bg-gray-700 dark:hover:border-gray-600",
    primary:
      "text-white bg-primary-700 hover:bg-primary-800 dark:bg-primary-600 dark:hover:bg-primary-700",
    purple:
      "text-white bg-purple-700 hover:bg-purple-800 dark:bg-purple-600 dark:hover:bg-purple-700",
    red: "text-white bg-red-700 hover:bg-red-800 dark:bg-red-600 dark:hover:bg-red-700",
    yellow: "text-white bg-yellow-400 hover:bg-yellow-500 ",
    none: "",
  }
  const colorCheckedClasses = {
    alternative:
      "text-primary-700 border dark:text-primary-500 bg-gray-100 dark:bg-gray-700 border-gray-300 shadow-gray-300 dark:shadow-gray-800 shadow-inner",
    blue: "text-blue-900 bg-blue-400 dark:bg-blue-500 shadow-blue-700 dark:shadow-blue-800 shadow-inner",
    dark: "text-white bg-gray-500 dark:bg-gray-600 shadow-gray-800 dark:shadow-gray-900 shadow-inner",
    green:
      "text-green-900 bg-green-400 dark:bg-green-500 shadow-green-700 dark:shadow-green-800 shadow-inner",
    light:
      "text-gray-900 bg-gray-100 border border-gray-300 dark:bg-gray-500 dark:text-gray-900 dark:border-gray-700 shadow-gray-300 dark:shadow-gray-700 shadow-inner",
    primary:
      "text-primary-900 bg-primary-400 dark:bg-primary-500 shadow-primary-700 dark:shadow-primary-800 shadow-inner",
    purple:
      "text-purple-900 bg-purple-400 dark:bg-purple-500 shadow-purple-700 dark:shadow-purple-800 shadow-inner",
    red: "text-red-900 bg-red-400 dark:bg-red-500 shadow-red-700 dark:shadow-red-800 shadow-inner",
    yellow:
      "text-yellow-900 bg-yellow-300 dark:bg-yellow-400 shadow-yellow-500 dark:shadow-yellow-700 shadow-inner",
    none: "",
  }
  const coloredFocusClasses = {
    alternative: "focus-within:ring-gray-200 dark:focus-within:ring-gray-700",
    blue: "focus-within:ring-blue-300 dark:focus-within:ring-blue-800",
    dark: "focus-within:ring-gray-300 dark:focus-within:ring-gray-700",
    green: "focus-within:ring-green-300 dark:focus-within:ring-green-800",
    light: "focus-within:ring-gray-200 dark:focus-within:ring-gray-700",
    primary: "focus-within:ring-primary-300 dark:focus-within:ring-primary-800",
    purple: "focus-within:ring-purple-300 dark:focus-within:ring-purple-900",
    red: "focus-within:ring-red-300 dark:focus-within:ring-red-900",
    yellow: "focus-within:ring-yellow-300 dark:focus-within:ring-yellow-900",
    none: "",
  }
  const coloredShadowClasses = {
    alternative: "shadow-gray-500/50 dark:shadow-gray-800/80",
    blue: "shadow-blue-500/50 dark:shadow-blue-800/80",
    dark: "shadow-gray-500/50 dark:shadow-gray-800/80",
    green: "shadow-green-500/50 dark:shadow-green-800/80",
    light: "shadow-gray-500/50 dark:shadow-gray-800/80",
    primary: "shadow-primary-500/50 dark:shadow-primary-800/80",
    purple: "shadow-purple-500/50 dark:shadow-purple-800/80",
    red: "shadow-red-500/50 dark:shadow-red-800/80 ",
    yellow: "shadow-yellow-500/50 dark:shadow-yellow-800/80 ",
    none: "",
  }
  const outlineClasses = {
    alternative:
      "text-gray-900 dark:text-gray-400 hover:text-white border border-gray-800 hover:bg-gray-900 focus-within:bg-gray-900 focus-within:text-white focus-within:ring-gray-300 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-600 dark:focus-within:ring-gray-800",
    blue: "text-blue-700 hover:text-white border border-blue-700 hover:bg-blue-800 dark:border-blue-500 dark:text-blue-500 dark:hover:text-white dark:hover:bg-blue-600",
    dark: "text-gray-900 hover:text-white border border-gray-800 hover:bg-gray-900 focus-within:bg-gray-900 focus-within:text-white dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-600",
    green:
      "text-green-700 hover:text-white border border-green-700 hover:bg-green-800 dark:border-green-500 dark:text-green-500 dark:hover:text-white dark:hover:bg-green-600",
    light:
      "text-gray-500 hover:text-gray-900 bg-white border border-gray-200 dark:border-gray-600 dark:hover:text-white dark:text-gray-400 hover:bg-gray-50 dark:bg-gray-700 dark:hover:bg-gray-600",
    primary:
      "text-primary-700 hover:text-white border border-primary-700 hover:bg-primary-700 dark:border-primary-500 dark:text-primary-500 dark:hover:text-white dark:hover:bg-primary-600",
    purple:
      "text-purple-700 hover:text-white border border-purple-700 hover:bg-purple-800 dark:border-purple-400 dark:text-purple-400 dark:hover:text-white dark:hover:bg-purple-500",
    red: "text-red-700 hover:text-white border border-red-700 hover:bg-red-800 dark:border-red-500 dark:text-red-500 dark:hover:text-white dark:hover:bg-red-600",
    yellow:
      "text-yellow-400 hover:text-white border border-yellow-400 hover:bg-yellow-500 dark:border-yellow-300 dark:text-yellow-300 dark:hover:text-white dark:hover:bg-yellow-400",
    none: "",
  }
  const sizeClasses = {
    xs: "px-3 py-2 text-xs",
    sm: "px-4 py-2 text-sm",
    md: "px-5 py-2.5 text-sm",
    lg: "px-5 py-3 text-base",
    xl: "px-6 py-3.5 text-base",
  }
  const hasBorder = () =>
    outline || color === "alternative" || color === "light"
  let buttonClass
  if ($$props.pill === void 0 && $$bindings.pill && pill !== void 0)
    $$bindings.pill(pill)
  if ($$props.outline === void 0 && $$bindings.outline && outline !== void 0)
    $$bindings.outline(outline)
  if ($$props.size === void 0 && $$bindings.size && size !== void 0)
    $$bindings.size(size)
  if ($$props.href === void 0 && $$bindings.href && href !== void 0)
    $$bindings.href(href)
  if ($$props.type === void 0 && $$bindings.type && type !== void 0)
    $$bindings.type(type)
  if ($$props.color === void 0 && $$bindings.color && color !== void 0)
    $$bindings.color(color)
  if ($$props.shadow === void 0 && $$bindings.shadow && shadow !== void 0)
    $$bindings.shadow(shadow)
  if ($$props.tag === void 0 && $$bindings.tag && tag !== void 0)
    $$bindings.tag(tag)
  if ($$props.checked === void 0 && $$bindings.checked && checked !== void 0)
    $$bindings.checked(checked)
  buttonClass = twMerge(
    "text-center font-medium",
    group ? "focus-within:ring-2" : "focus-within:ring-4",
    group && "focus-within:z-10",
    group || "focus-within:outline-none",
    "inline-flex items-center justify-center " + sizeClasses[size],
    outline && checked && "border dark:border-gray-900",
    outline && checked && colorCheckedClasses[color],
    outline && !checked && outlineClasses[color],
    !outline && checked && colorCheckedClasses[color],
    !outline && !checked && colorClasses[color],
    color === "alternative" &&
      (group && !checked
        ? "dark:bg-gray-700 dark:text-white dark:border-gray-700 dark:hover:border-gray-600 dark:hover:bg-gray-600"
        : "dark:bg-transparent dark:border-gray-600 dark:hover:border-gray-700"),
    outline &&
      color === "dark" &&
      (group
        ? checked
          ? "bg-gray-900 border-gray-800 dark:border-white dark:bg-gray-600"
          : "dark:text-white border-gray-800 dark:border-white"
        : "dark:text-gray-400 dark:border-gray-700"),
    coloredFocusClasses[color],
    hasBorder() && group && "border-s-0 first:border-s",
    group
      ? (pill && "first:rounded-s-full last:rounded-e-full") ||
          "first:rounded-s-lg last:rounded-e-lg"
      : (pill && "rounded-full") || "rounded-lg",
    shadow && "shadow-lg",
    shadow && coloredShadowClasses[color],
    $$props.disabled && "cursor-not-allowed opacity-50",
    $$props.class,
  )
  return `${
    href
      ? `<a${spread(
          [
            { href: escape_attribute_value(href) },
            escape_object($$restProps),
            {
              class: escape_attribute_value(buttonClass),
            },
            { role: "button" },
          ],
          {},
        )}>${slots.default ? slots.default({}) : ``}</a>`
      : `${
          tag === "button"
            ? `<button${spread(
                [
                  { type: escape_attribute_value(type) },
                  escape_object($$restProps),
                  {
                    class: escape_attribute_value(buttonClass),
                  },
                ],
                {},
              )}>${slots.default ? slots.default({}) : ``}</button>`
            : `${(tag$1 => {
                return tag$1
                  ? `<${tag}${spread(
                      [
                        escape_object($$restProps),
                        {
                          class: escape_attribute_value(buttonClass),
                        },
                      ],
                      {},
                    )}>${is_void(tag$1) ? "" : `${slots.default ? slots.default({}) : ``}`}${is_void(tag$1) ? "" : `</${tag$1}>`}`
                  : ""
              })(tag)}`
        }`
  } `
})
export { Button as B }
