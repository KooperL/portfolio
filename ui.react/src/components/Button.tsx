import { Scheme } from "../containers/context/type";

export function Button(props: {colours: Scheme, disabled?: Boolean}) {
  const style = {
    backgroundColor: props.colours.button.bgSolid,
    color: props.colours.button.text,
    borderRadius: '10px',
    padding: '0.2rem 2rem',
    // margin: '10px'
  } as const;
  return (
    <button className="submit-button" type='submit' name='submit' value='Submit' style={style}>Submit</button>
  );
}

// https://stackoverflow.com/questions/28365233/inline-css-styles-in-react-how-to-implement-ahover