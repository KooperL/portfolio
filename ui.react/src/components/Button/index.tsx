import { Scheme } from "../../containers/context/type"

export function Button(props: {
  colours: Scheme
  disabled?: Boolean
  callBack?: () => void
  label?: string
  action?: "button" | "submit" | "reset" | undefined
}) {
  const style = {
    backgroundColor: props.colours.button.bgSolid,
    color: props.colours.button.text,
    borderRadius: "10px",
    padding: "0.2rem 2rem",
    // margin: '10px'
  } as const
  if (props.callBack) {
    return (
      <div className="submit-button-container">
        {/* @ts-ignore - Sooo bad.... cmon ts, it exists!!!!  */}
        <button
          className="submit-button"
          value={props.label ?? "submit"}
          style={style}
          onClick={() => {
            props.callBack && props.callBack()
          }}
        >
          {props.label ?? "Submit"}
        </button>
      </div>
    )
  }
  return (
    <button
      className="submit-button"
      type={props.action ?? "submit"}
      name="submit"
      value={props.label ?? "Submit"}
      style={style}
    >
      {props.label ?? "Submit"}
    </button>
  )
}

// https://stackoverflow.com/questions/28365233/inline-css-styles-in-react-how-to-implement-ahover
