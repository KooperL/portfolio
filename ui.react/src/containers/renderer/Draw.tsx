import { triangle } from "./utils/types";

export function DrawFigure(props: {name: string, h: number | string, w: number | string}) {
  // const displayGCD = gcd(+props.w, +props.h);
  return (
    <figure>
      {/* <svg viewBox={`${0-+props.w/2} ${0-+props.h/2} ${+props.w/2} ${+props.h/2}`} width={props.w} height={props.h}> */}
      <svg viewBox={`0 0 ${+props.w} ${+props.h}`} width={props.w} height={props.h}>
        <use xlinkHref={`#${props.name}`}/>
      </svg>
    </figure>
  );
}

export function drawBlueprint(props: {name: string, tris: triangle[]}) {
  return (
    <svg className="defs-only" xmlns="http://www.w3.org/2000/svg">
      <symbol id={props.name}>
        {props.tris.map((triangle, triangleIndex) => (
          <polygon key={triangleIndex}
            points={`${triangle.p[0].x},${triangle.p[0].y} ${triangle.p[1].x},${triangle.p[1].y} ${triangle.p[2].x},${triangle.p[2].y}`}
            style={{
              fill: triangle['c']['fill'] ?? 'red',
              stroke: triangle['c']['stroke'] ?? 'pink',
              strokeWidth: triangle['c']['strokeWidth'] || 5.0,
            }}
          />
          ))}
      </symbol>
    </svg>
  );
}