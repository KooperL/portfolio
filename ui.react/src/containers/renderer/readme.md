Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Use ticker as switch to toggle different transformations for grid
 - Do the scale properly -> make it a method and take args
 - colour spectrum