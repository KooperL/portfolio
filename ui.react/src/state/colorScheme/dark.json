{
  "header": {
    "background": "#205375",
    "text": "#EFEFEF",
    "outline": "#EFEFEF"
  },
  "body": {
    "background": "#112B3C",
    "foreground": "#205375",
    "h1": "#F66B0E",
    "h2": "#EFEFEF",
    "text": "#F2F2F2"
  },
  "button": {
    "bgSolid": "#205375",
    "bgGrad1": "#455375",
    "bgGrad2": "#20535a",
    "text": "#F2F2F2"
  }
}
