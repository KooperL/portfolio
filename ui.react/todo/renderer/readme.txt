Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Put engine components behind layers of abstraction to be changed without affecting core engine -> make it a method and take args
 - colour spectrum
 - switch between canvas, web-gl enabled canvas and svg rasterisation

Abandoning, performance is simply not going to be acceptable without spending more time learning optimising strategies. Time better spent implementing algo's with a public library. Shame.
KT to myself:
 - V1 had best performance, but function structure was a mess and wasn't designed to be a general engine
 - V2 Was more well made and class based, not optimal for react but function structure was very ideal. Probably where I should pick up
 - V3 Was V2 crammed into react functional components, eventually started impacting performance.