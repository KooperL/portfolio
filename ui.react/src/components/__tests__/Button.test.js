import { Button } from '../Button'
import React from 'react'
import renderer from 'react-test-renderer'
import dark from '../../state/colorScheme/dark.json'

describe('Button component', () => {
  test('it matches the snapshot', () => {
    const tree = renderer.create(<Button colours={dark} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
