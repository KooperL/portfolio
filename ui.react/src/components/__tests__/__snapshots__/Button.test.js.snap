// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component it matches the snapshot 1`] = `
<button
  className="submit-button"
  name="submit"
  style={
    Object {
      "backgroundColor": "#205375",
      "borderRadius": "10px",
      "color": "#F2F2F2",
      "padding": "0.2rem 2rem",
    }
  }
  type="submit"
  value="Submit"
>
  Submit
</button>
`;
